### DISCUSSION TITLE
Cloning app into different folders

### DISCUSSION THREAD

**Discussion by yahiakala** [benlawraus/anvil_works_design_test#1] [Q&A] [answered]
Hi, great project! I have a question about the folder structure. I noticed that in git_pull_from_anvil_works.zsh, you copy some of the contents from the actual anvil app repo into the main (higher level) directory. Similarly, on the push script, code is synced from the main directory into the app repo before pushing. Is this so that the local development happens in the main directory and the mock classes work?

In other words, I don't touch anything that is in the cloned anvil app repo?

**Comment by benlawraus** [comment, marked as answer]
Hi
Thanks for your feedback.
I did this because:
1. I could have two repos for the anvil code, one for the project and one for the anvil.works. 
2. Sometimes I make so many mistakes while I am debugging, it is nice just to copy the original code back easily.  (I also put them in the backup folder which may be going too far)
3. there are files in the project anvil code that are not in the anvil.works code specifically: _anvil.designer.py 

There might be other reasons, that I have forgotten. But in the end, they are synced in the scripts of git_pull_.. and git_push... except for the _anvil_desigher.py files

Anyway, you are welcome to change the git_pull and git_push scripts. ChatGPT is a nice help if you are on windows.
regards
Ben
